### PR TITLE
Less verbose cmake_find_package

### PR DESCRIPTION
**Pull request by Ignition**
Changelog: Fix: cmake_find_package no longer seeks to find packages which are already found.
Docs: Omit

find_package would repeat messages for common packages within the dependencies DAG. This commit skips the dependency find_package if the package was already found.

### CODE REVIEW

**Comment by czoido** [comment]
Hi @Ignition,
I have added a test for the feature as an example and solved the conflicts. We would like to merge this for 1.24 that's going to be released at the end of the March. 
Could you please check the comments above?
Thanks a lot.

File: conans/test/functional/generators/cmake_find_package_test.py [hunk 1]
```diff
@@ -105,6 +106,42 @@ def cmake_lock_target_redefinition_test(self):
...
+            settings = "os", "arch", "compiler"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+
+        """)
+
+        cmakelists = textwrap.dedent("""
+        cmake_minimum_required(VERSION 3.0)
```
Comment (memsharded) [inline conans/test/functional/generators/cmake_find_package_test.py]: indent file contents

**Comment by jgsogo** [comment]
Tested this with `poco` and `boost`. The only repeated package there is `zlib`:

image

And the lines corresponding to `zlib` appear several times too:

```
-- Conan: Using autogenerated FindZLIB.cmake
-- Library z found /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
-- Skipping already existing target: CONAN_LIB::ZLIB_z
-- Found: /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
```

Full log:

```
-- Conan: Using autogenerated FindBoost.cmake
-- Library boost_wave found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_wave.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_wave.a
-- Library boost_container found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_container.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_container.a
-- Library boost_contract found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_contract.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_contract.a
-- Library boost_exception found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_exception.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_exception.a
-- Library boost_graph found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_graph.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_graph.a
-- Library boost_iostreams found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_iostreams.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_iostreams.a
-- Library boost_locale found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_locale.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_locale.a
-- Library boost_log found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_log.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_log.a
-- Library boost_program_options found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_program_options.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_program_options.a
-- Library boost_random found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_random.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_random.a
-- Library boost_regex found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_regex.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_regex.a
-- Library boost_serialization found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_serialization.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_serialization.a
-- Library boost_wserialization found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_wserialization.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_wserialization.a
-- Library boost_coroutine found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_coroutine.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_coroutine.a
-- Library boost_context found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_context.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_context.a
-- Library boost_timer found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_timer.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_timer.a
-- Library boost_thread found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_thread.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_thread.a
-- Library boost_chrono found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_chrono.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_chrono.a
-- Library boost_date_time found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_date_time.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_date_time.a
-- Library boost_atomic found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_atomic.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_atomic.a
-- Library boost_filesystem found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_filesystem.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_filesystem.a
-- Library boost_system found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_system.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_system.a
-- Library boost_type_erasure found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_type_erasure.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_type_erasure.a
-- Library boost_log_setup found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_log_setup.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_log_setup.a
-- Library boost_math_c99 found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99.a
-- Library boost_math_c99f found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99f.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99f.a
-- Library boost_math_c99l found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99l.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_c99l.a
-- Library boost_math_tr1 found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1.a
-- Library boost_math_tr1f found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1f.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1f.a
-- Library boost_math_tr1l found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1l.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_math_tr1l.a
-- Library boost_stacktrace_addr2line found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_addr2line.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_addr2line.a
-- Library boost_stacktrace_basic found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_basic.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_basic.a
-- Library boost_stacktrace_noop found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_noop.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_stacktrace_noop.a
-- Library boost_unit_test_framework found /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_unit_test_framework.a
-- Found: /Users/jgsogo/.conan/data/boost/1.72.0/_/_/package/69168f775732984eb37d785004b6ef25111fe5f9/lib/libboost_unit_test_framework.a
-- Conan: Using autogenerated FindZLIB.cmake
-- Library z found /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
-- Found: /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
-- Conan: Using autogenerated FindBZip2.cmake
-- Library bz2 found /Users/jgsogo/.conan/data/bzip2/1.0.8/_/_/package/32bef4803d4b079e983ecb27f105881e778bc5a7/lib/libbz2.a
-- Found: /Users/jgsogo/.conan/data/bzip2/1.0.8/_/_/package/32bef4803d4b079e983ecb27f105881e778bc5a7/lib/libbz2.a
-- Conan: Using autogenerated FindPoco.cmake
-- Library PocoMongoDB found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoMongoDB.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoMongoDB.a
-- Library PocoNetSSL found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoNetSSL.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoNetSSL.a
-- Library PocoNet found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoNet.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoNet.a
-- Library PocoCrypto found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoCrypto.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoCrypto.a
-- Library PocoDataSQLite found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoDataSQLite.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoDataSQLite.a
-- Library PocoData found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoData.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoData.a
-- Library PocoZip found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoZip.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoZip.a
-- Library PocoUtil found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoUtil.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoUtil.a
-- Library PocoXML found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoXML.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoXML.a
-- Library PocoJSON found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoJSON.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoJSON.a
-- Library PocoRedis found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoRedis.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoRedis.a
-- Library PocoFoundation found /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoFoundation.a
-- Found: /Users/jgsogo/.conan/data/poco/1.10.1/_/_/package/8150a577f628df026b4ed5cf8cb5abcf1f60ca9b/lib/libPocoFoundation.a
-- Conan: Using autogenerated FindOpenSSL.cmake
-- Library ssl found /Users/jgsogo/.conan/data/openssl/1.0.2t/_/_/package/eb50d18a5a5d59bd0c332464a4c348ab65e353bf/lib/libssl.a
-- Found: /Users/jgsogo/.conan/data/openssl/1.0.2t/_/_/package/eb50d18a5a5d59bd0c332464a4c348ab65e353bf/lib/libssl.a
-- Library crypto found /Users/jgsogo/.conan/data/openssl/1.0.2t/_/_/package/eb50d18a5a5d59bd0c332464a4c348ab65e353bf/lib/libcrypto.a
-- Found: /Users/jgsogo/.conan/data/openssl/1.0.2t/_/_/package/eb50d18a5a5d59bd0c332464a4c348ab65e353bf/lib/libcrypto.a
-- Conan: Using autogenerated FindZLIB.cmake
-- Library z found /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
-- Skipping already existing target: CONAN_LIB::ZLIB_z
-- Found: /Users/jgsogo/.conan/data/zlib/1.2.11/_/_/package/f74366f76f700cc6e991285892ad7a23c30e6d47/lib/libz.a
```

**Comment by memsharded** [comment]
Ok, lets put it into Conan 1.24. Thanks!

**Comment by Ignition** [comment]
Sorry for not responding till now, been on holiday (not working/looking at email). Thanks for continuing to improve this PR and merging.

**Pull request merged**